apps/root/src/pages/position-detail/components/position-summary-controls/index.tsx

```diff
@@ -369,162 +367,167 @@ const PositionSummaryControls = ({ pendingTransaction, position }: PositionSumma
+        onClick: () => onWithdraw(!!hasSignSupport && position.to.address === PROTOCOL_TOKEN_ADDRESS),
+      },
+      ...(shouldShowWithdrawWrappedToken
+        ? [
+            {
+              type: OptionsMenuOptionType.option,
+              label: intl.formatMessage(
+                defineMessage({
+                  description: 'withdrawWrapped',
+                  defaultMessage: 'Withdraw {wrappedProtocolToken}',
+                }),
+                {
+                  wrappedProtocolToken: wrappedProtocolToken.symbol,
+                }
+              ),
+              disabled: disabledWithdraw || isPending || disabled,
+              onClick: () => onWithdraw(false),
+            },
+          ]
+        : []),
+      {
+        type: OptionsMenuOptionType.option,
+        label: (
+          <FormattedMessage
+            description="withdraw funds"
+            defaultMessage="Withdraw remaining {token}"
+            values={{ token: position.from.symbol }}
+          />
+        ),
+        disabled: disabledWithdrawFunds || isPending || disabled || position.remainingLiquidity.amount <= 0n,
+        onClick: onWithdrawFunds,
+      },
+    ],
+    [intl, wrappedProtocolToken, disabledWithdraw, isPending, disabled, position, hasSignSupport]
+  );
 
   return (
-    <>
-      <TerminateModal open={showTerminateModal} position={position} onCancel={() => setShowTerminateModal(false)} />
-      <ModifySettingsModal
-        open={showModifyRateSettingsModal}
-        position={position}
-        onCancel={() => setShowModifyRateSettingsModal(false)}
-      />
-      <TransferPositionModal
-        open={showTransferModal}
-        position={position}
-        onCancel={() => setShowTransferModal(false)}
-      />
-      <NFTModal open={showNFTModal} nftData={nftData} onCancel={() => setShowNFTModal(false)} />
-      <ContainerBox gap={3} alignSelf="end">
-        {showExtendedFunctions && (
-          <Button variant="outlined" disabled={disableModifyPosition} onClick={onModifyRate}>
-            <FormattedMessage description="managePosition" defaultMessage="Manage position" />
-          </Button>
-        )}
-
-        {shouldDisableArrow && (
-          <Button
-            variant="outlined"
-            disabled={disabledWithdraw || isPending || disabled || position.toWithdraw.amount <= 0n}
-            onClick={() => onWithdraw(!!hasSignSupport && position.to.address === PROTOCOL_TOKEN_ADDRESS)}
-          >
-            <FormattedMessage
-              description="withdrawToken"
-              defaultMessage="Withdraw {token}"
-              values={{
-                token:
-                  hasSignSupport || position.to.address !== PROTOCOL_TOKEN_ADDRESS
-                    ? position.to.symbol
-                    : wrappedProtocolToken.symbol,
-              }}
-            />
-          </Button>
-        )}
+    ownerWallet && (
```

Comment:
We could check this before even rendering the component if we are just going to return nothing if the wallet is not there